spec/rubocop/cop/style/guard_clause_spec.rb

```diff
@@ -323,6 +323,33 @@ def func
     RUBY
   end
 
+  it 'registers an offense when using lvar as an argument of raise in `else` branch' do
+    expect_offense(<<~RUBY)
+      def func
+      rescue => e
+        unless condition
+        ^^^^^^ Use a guard clause (`raise e if condition`) instead of wrapping the code inside a conditional expression.
+          retry
+        else
+          raise e
+        end
+      end
+    RUBY
+
+    # NOTE: Let `Layout/HeredocIndentation`, `Layout/ClosingHeredocIndentation`, and
+    #       `Layout/IndentationConsistency` cops autocorrect inconsistent indentations.
```

Comment:
heredoc is not used. Probably `Layout/TrailingWhitespace`, `Layout/EmptyLines`, and `Layout/EmptyLinesAroundMethodBody` cops.